sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -0,0 +1,252 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+
+from sklearn.preprocessing import TargetRegressorEncoder
+
+
+@pytest.mark.parametrize('categories, unknown_X', [
```

Comment:
Looks like `unknown_X` does not need to be parametrized as long as it's of dtype object

Also `X_with_unknown` might be a more accurate name.

Reply:
same for other checks below